synapse/handlers/sso.py

```diff
@@ -119,15 +120,16 @@ async def get_sso_user_by_remote_user_id(
         # No match.
         return None
 
-    async def get_mxid_from_sso(
+    async def complete_sso_login_request(
         self,
         auth_provider_id: str,
         remote_user_id: str,
-        user_agent: str,
-        ip_address: str,
+        request: SynapseRequest,
+        client_redirect_url: str,
         sso_to_matrix_id_mapper: Callable[[int], Awaitable[UserAttributes]],
         grandfather_existing_users: Optional[Callable[[], Awaitable[Optional[str]]]],
-    ) -> str:
+        extra_login_attributes: Optional[JsonDict],
```

Comment:
Might make sense to have `extra_login_attributes` default to `None`. 🤷 This method seems to be growing, but it is a nice de-duplication!

Reply:
:+1: